beacon_node/beacon_chain/src/fetch_blobs/mod.rs

```diff
@@ -378,26 +385,13 @@ async fn compute_custody_columns_to_import<T: BeaconChainTypes>(
                 if let Some(known_columns) =
                     chain_adapter_cloned.cached_data_column_indexes(&block_root)
                 {
-                    custody_columns.retain(|col| !known_columns.contains(&col.index));
+                    custody_columns.retain(|col| !known_columns.contains(&col.index()));
                     if custody_columns.is_empty() {
                         return Ok(vec![]);
                     }
                 }
 
-                // KZG verify data columns before publishing. This prevents blobs with invalid
-                // KZG proofs from the EL making it into the data availability checker. We do not
-                // immediately add these blobs to the observed blobs/columns cache because we want
-                // to allow blobs/columns to arrive on gossip and be accepted (and propagated) while
-                // we are waiting to publish. Just before publishing we will observe the blobs/columns
-                // and only proceed with publishing if they are not yet seen.
```

Comment:
Should we keep the part of this comment about not marking data columns as seen here, while we wait to publish? AFAICT it is still relevant as publishing happens here:

https://github.com/sigp/lighthouse/blob/d7ab88cf858ec90696b1dbba199af34823d20e61/beacon_node/beacon_chain/src/fetch_blobs/mod.rs#L314-L316

The `publish_fn` in practice is:

https://github.com/sigp/lighthouse/blob/cfb1f7331064b758c6786e4e1dc15507af5ff5d1/beacon_node/network/src/network_beacon_processor/mod.rs#L1014-L1020

Which does still do the observation:

https://github.com/sigp/lighthouse/blob/cfb1f7331064b758c6786e4e1dc15507af5ff5d1/beacon_node/network/src/network_beacon_processor/mod.rs#L1057

Reply:
Nice, added in https://github.com/sigp/lighthouse/pull/7713/commits/c277d5092f176fc0271b9d808ac9850349446368. Thanks!